validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java

```diff
@@ -0,0 +1,120 @@
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final StringValueTypeDefinition<BLSPublicKey> PUBKEY_TYPE =
+      DeserializableTypeDefinition.string(BLSPublicKey.class)
+          .formatter(BLSPublicKey::toString)
+          .parser(BLSPublicKey::fromHexString)
+          .format("byte")
+          .build();
+  private static final DeserializableTypeDefinition<Config> CONFIG_TYPE =
+      DeserializableTypeDefinition.object(Config.class)
+          .initializer(Config::new)
+          .name("RuntimeProposerConfig")
+          .withField(
+              "fee_recipient", ETH1ADDRESS_TYPE, Config::getFeeRecipient, Config::setFeeRecipient)
+          .build();
+  private static final DeserializableTypeDefinition<Map<BLSPublicKey, Config>> CONFIG_MAP_TYPE =
+      DeserializableTypeDefinition.mapOf(PUBKEY_TYPE, CONFIG_TYPE, ConcurrentHashMap::new);
+
+  private final Map<BLSPublicKey, Config> proposerConfigMap = new ConcurrentHashMap<>();
+
+  public RuntimeProposerConfig(final Optional<Path> storagePath) {
+    this.storagePath = storagePath;
+    storagePath.ifPresent(this::read);
+  }
+
+  public Optional<Eth1Address> getEth1AddressForPubKey(final BLSPublicKey pubKey) {
+    final Config config = proposerConfigMap.get(pubKey);
+    if (config == null) {
+      return Optional.empty();
+    }
+    return Optional.of(config.getFeeRecipient());
+  }
+
+  public synchronized void addOrUpdate(
+      final BLSPublicKey publicKey, final Eth1Address eth1Address) {
+    proposerConfigMap.put(publicKey, new Config(eth1Address));
+    storagePath.ifPresent(this::save);
+  }
+
+  public void delete(final BLSPublicKey publicKey) {
+    proposerConfigMap.remove(publicKey);
+    storagePath.ifPresent(this::save);
+  }
+
+  private void save(final Path path) {
+    try (OutputStream writer = new FileOutputStream(path.toFile(), false)) {
+      JsonUtil.serializeToBytes(proposerConfigMap, CONFIG_MAP_TYPE, writer);
+    } catch (IOException e) {
+      LOG.error("Failed to store file: " + path.toAbsolutePath(), e);
+    }
+  }
+
+  private synchronized void read(final Path path) {
+    if (!path.toFile().exists()) {
+      return;
+    }
+    if (!proposerConfigMap.isEmpty()) {
+      proposerConfigMap.clear();
+    }
+    try (InputStream inputStream = new FileInputStream(path.toFile())) {
+      proposerConfigMap.putAll(JsonUtil.parse(inputStream, CONFIG_MAP_TYPE));
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to parse file: " + path.toAbsolutePath(), e);
+    }
+  }
+
+  static class Config {
+    private Eth1Address feeRecipient;
+
+    public Config() {}
+
+    public Config(final Eth1Address feeRecipient) {
+      this.feeRecipient = feeRecipient;
+    }
+
+    public Eth1Address getFeeRecipient() {
+      return feeRecipient;
+    }
+
+    public void setFeeRecipient(final Eth1Address feeRecipient) {
+      this.feeRecipient = feeRecipient;
+    }
```

Comment:
Worth noting that this isn't thread safe which is potentially dangerous when it's being using in a `ConcurrentHashMap`.  Makes sense for parsing but probably worth a comment to make it clear that it can only be used for parsing.

Reply:
changed to use a builder